itsy-bitsy-data-structures.js

```diff
@@ -1380,13 +1380,16 @@ class BinarySearchTree {
     if (this.root === null) {
       this.root = node;
       return;
+
+    // If the value already exists in the tree, we don't do anyting.
+    } else if (this.contains(value)) {
+      return;
     }
 
     // We start at the root.
     var current = this.root;
 
-    // We're going to loop until we've either added our item or discovered it
-    // already exists in the tree.
```

Comment:
again really helpful for someone to understand what is happening here without understanding the code on first pass.